qiskit_ibm_runtime/qpy/binary_io/circuits.py

```diff
@@ -0,0 +1,1288 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Binary IO for circuit objects."""
+
+import io
+import json
+import struct
+import uuid
+import warnings
+
+from collections import defaultdict
+import numpy as np
+
+from qiskit import circuit as circuit_mod
+from qiskit import extensions
+from qiskit.circuit import library, controlflow, CircuitInstruction, ControlFlowOp
+from qiskit.circuit.classical import expr
+from qiskit.circuit.classicalregister import ClassicalRegister, Clbit
+from qiskit.circuit.gate import Gate
+from qiskit.circuit.singleton import SingletonInstruction, SingletonGate
+from qiskit.circuit.controlledgate import ControlledGate
+from qiskit.circuit.instruction import Instruction
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit.quantumregister import QuantumRegister, Qubit
+from qiskit.extensions import quantum_initializer
```

Comment:
Note that `qiskit.extensions` has been removed from Qiskit's `main` branch and will not be in Qiskit 1.0. This same line in qiskit-ibm-provider currently causes `import qiskit_ibm_provider` to fail when you have Qiskit `main` installed. I was going to open an issue for qiskit-ibm-provider but since the code is moving here I thought I would just mention it here.

Reply:
Thanks for bringing this up - we copy the qpy module directly from qiskit so i'll make sure to handle that in a separate PR 

Created an issue https://github.com/Qiskit/qiskit-ibm-runtime/issues/1338